zio-interop-cats-tests/shared/src/test/scala/zio/interop/CatsSpecBase.scala

```diff
@@ -113,44 +117,62 @@ private[zio] trait CatsSpecBase
   implicit val eqForNothing: Eq[Nothing] =
     Eq.allEqual
 
+  // workaround for laws `evalOn local pure` & `executionContext commutativity`
+  // (ZIO cannot implement them at all due to `.executor.asEC` losing the original executionContext)
   implicit val eqForExecutionContext: Eq[ExecutionContext] =
     Eq.allEqual
 
   implicit val eqForCauseOfNothing: Eq[Cause[Nothing]] =
-    (x, y) => (x.isInterrupted && y.isInterrupted) || x == y
-
-  implicit def eqForExitOfNothing[A: Eq]: Eq[Exit[Nothing, A]] = {
-    case (Exit.Success(x), Exit.Success(y)) => x eqv y
-    case (Exit.Failure(x), Exit.Failure(y)) => x eqv y
-    case _                                  => false
-  }
+    (x, y) => (x.isInterrupted && y.isInterrupted && x.failureOption.isEmpty && y.failureOption.isEmpty) || x == y
 
   implicit def eqForUIO[A: Eq](implicit ticker: Ticker): Eq[UIO[A]] = { (uio1, uio2) =>
-    val exit1 = unsafeRun(uio1)
-    val exit2 = unsafeRun(uio2)
-    (exit1 eqv exit2) || {
-      println(s"$exit1 was not equal to $exit2")
-      false
+    val (exit1, i1) = unsafeRun(uio1)
+    val (exit2, i2) = unsafeRun(uio2)
+    val out1        = toOutcomeCauseOtherFiber[Id, Nothing, Option[A]](i1)(identity, exit1)
+    val out2        = toOutcomeCauseOtherFiber[Id, Nothing, Option[A]](i2)(identity, exit2)
+    (out1, out2) match {
+      case (Succeeded(Some(a)), Succeeded(Some(b)))          => a eqv b
+      case (Succeeded(Some(_)), _) | (_, Succeeded(Some(_))) =>
+        println(s"$out1 was not equal to $out2")
+        false
+      case _                                                 => true
```

Comment:
We can ignore results like `Succeeded(None)`, it means that IO under tests did not produce a result (`IO.never`, for example)

Reply:
@johnspade We can't ignore it. If both lhs and rhs hang then it's fine, but not if one of the sides finishes and the other hangs. And `case _ => true` here ignores everything, not just `Succeeded(None)`

Reply:
Well, cats-effect uses similar relaxed `Eq` for law testing

Reply:
@johnspade
I would *really* love to know how you came up with that, but cats-effect tests use the `eqIOA` instance defined in `TestInstances` - https://github.com/typelevel/cats-effect/blob/series/3.x/tests/shared/src/test/scala/cats/effect/IOSpec.scala#L1580 - which uses the `Eq[Outcome[Option, Throwable, A]]` instance - i.e. `out1 eqv out2` here - which checks all the branches not just success.

```scala
  implicit def eq[F[_], E: Eq, A](implicit FA: Eq[F[A]]): Eq[Outcome[F, E, A]] =
    Eq instance {
      case (Canceled(), Canceled()) => true
      case (Errored(left), Errored(right)) => left === right
      case (Succeeded(left), Succeeded(right)) => left === right
      case _ => false
    }
```